relayer/chain/ethereum/listener.go

```diff
@@ -117,23 +140,64 @@ func (li *Listener) pollEventsAndHeaders(
 			}
 
 			finalizedBlockNumber := gethheader.Number.Uint64() - descendantsUntilFinal
-			var events []*outbound.ContractMessage
-
+			var channelEvents []*outbound.ContractMessage
 			for _, channelContract := range li.contracts {
-				channelEvents, err := li.queryEvents(ctx, channelContract, finalizedBlockNumber, &finalizedBlockNumber)
+				contractEvents, err := li.queryChannelEvents(ctx, channelContract, finalizedBlockNumber, &finalizedBlockNumber)
 				if err != nil {
 					li.log.WithError(err).Error("Failure fetching event logs")
 				}
 
-				events = append(events, channelEvents...)
+				channelEvents = append(channelEvents, contractEvents...)
+			}
+			li.forwardChannelEvents(ctx, hcs, channelEvents)
+
+			// Query PolkadotRelayChainBridge contract's InitialVerificationSuccessful events
+			blockNumber := gethheader.Number.Uint64()
+			var events []*polkadotrelaychainbridge.ContractInitialVerificationSuccessful
+
+			contractEvents, err := li.queryLightClientEvents(ctx, blockNumber, &blockNumber)
+			if err != nil {
+				li.log.WithError(err).Error("Failure fetching event logs")
 			}
+			events = append(events, contractEvents...)
 
-			li.forwardEvents(ctx, hcs, events)
+			if len(events) > 0 {
+				li.log.Info(fmt.Sprintf("Found %d PolkadotRelayChainBridge contract events on block %d", len(events), blockNumber))
+			}
+			li.processLightClientEvents(ctx, events)
+
+			// Mark items ReadyToComplete if the current block number has passed their CompleteOnBlock number
+			items := li.db.GetItemsByStatus(store.InitialVerificationTxConfirmed)
+			if len(items) > 0 {
+				li.log.Info(fmt.Sprintf("Found %d InitialVerificationTxConfirmed items", len(items)))
+			}
+			for _, item := range items {
+				if item.CompleteOnBlock <= blockNumber {
+					li.log.Info("4: Updating item status from 'InitialVerificationTxConfirmed' to 'ReadyToComplete'")
+
+					// Fetch intended completion block's hash
+					blockHash := gethheader.Hash()
+					if item.CompleteOnBlock > blockNumber {
```

Comment:
seems weird, first `if item.CompleteOnBlock <= blockNumber` then `if item.CompleteOnBlock > blockNumber` - does this block ever run?

Reply:
Yeah second conditional should be `if item.CompleteOnBlock < blockNumber`.

Reply:
Updated in https://github.com/Snowfork/polkadot-ethereum/pull/314/commits/bc765754f8d0e0df068a2b39d0e6bf1225b6eb18